processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -378,6 +379,13 @@ else if ( !method.isUpdateMethod() ) {
 
             }
 
+            Type subclassExhaustiveExceptionType = ctx.getTypeFactory().getType( IllegalArgumentException.class );
+            TypeMirror subclassExhaustiveException =
+                beanMapping != null ? beanMapping.getSubclassExhaustiveException() : null;
+            if ( subclassExhaustiveException != null ) {
+                subclassExhaustiveExceptionType = ctx.getTypeFactory().getType( subclassExhaustiveException );
+            }
```

Comment:
This can be simplified:

```suggestion
            TypeMirror subclassExhaustiveException = method.getOptions()
                .getBeanMapping()
                .getSubclassExhaustiveException();
            Type subclassExhaustiveExceptionType = ctx.getTypeFactory().getType( subclassExhaustiveException );
```

Reply:
It can indeed be simplified here, but does `beanMapping `need to be judged `null`? Because I saw that other places in the code that use `beanMapping ` are judged null.

Reply:
I think you were looking in some of the lower level places where we are accessing the annotation. I see that on line 171 it is checked for `null`. However, what I proposed is already used on line 335, so I am not worried about that.